test/testgarbage.cpp

```diff
@@ -548,11 +548,10 @@ class TestGarbage : public TestFixture {
 
     void garbageCode23() {
         //garbage code : don't crash (#3481)
-        checkCode("{\n"
-                  "    if (1) = x\n"
-                  "    else abort s[2]\n"
-                  "}");
-        ASSERT_EQUALS("", errout.str());
+        ASSERT_THROW(checkCode("{\n"
+                               "    if (1) = x\n"
+                               "    else abort s[2]\n"
+                               "}"), InternalError);
```

Comment:
hmm.. in this specific code it would be better to throw syntaxError than unknownMacro. In the raw code an else can be followed by variables/functions call/etc.
maybe we could throw a syntax error if there is not a `;|}` before the else also?